### DISCUSSION TITLE
Extend `weapon` functionality

### DISCUSSION THREAD

**Discussion by vurvdev** [wiremod/wire#2677] [Suggestions]
1. Events that correspond to https://wiki.facepunch.com/gmod/GM:WeaponEquip and https://wiki.facepunch.com/gmod/GM:PlayerSwitchWeapon
2. Give weapon method
3. Setters for ammo
4. Weapon select method

etc

**Comment by Denneisk** [comment]
~~How do you supposed restriction of weapon giving and ammo setters should go?~~ Never mind didn't see a PR was already up for this.

**Comment by vurvdev** [comment]
Resolved by https://github.com/wiremod/wire/pull/2683
